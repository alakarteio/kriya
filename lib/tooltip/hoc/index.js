'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _tooltip = require('./tooltip');

Object.defineProperty(exports, 'default', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_tooltip).default;
  }
});

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }