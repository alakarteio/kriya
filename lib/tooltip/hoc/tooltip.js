'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames2 = require('classnames');

var _classnames3 = _interopRequireDefault(_classnames2);

var _tooltip = require('../tooltip');

var _tooltip2 = _interopRequireDefault(_tooltip);

var _tooltipStyles = require('../../../src/tooltip/hoc/tooltip.styles.scss');

var _tooltipStyles2 = _interopRequireDefault(_tooltipStyles);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

exports.default = function () {
  var config = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {
    left: true,
    top: false,
    right: false,
    bottom: false,
    className: ''
  };
  return function (text) {
    return function (Component) {
      return function (props) {
        var _classnames;

        var left = config.left,
            top = config.top,
            right = config.right,
            bottom = config.bottom,
            className = config.className;

        var classes = (0, _classnames3.default)(_tooltipStyles2.default.component, className);

        var tooltipClasses = (0, _classnames3.default)(_tooltipStyles2.default.tooltip, (_classnames = {}, _defineProperty(_classnames, _tooltipStyles2.default.left, left), _defineProperty(_classnames, _tooltipStyles2.default.top, top), _defineProperty(_classnames, _tooltipStyles2.default.right, right), _defineProperty(_classnames, _tooltipStyles2.default.bottom, bottom), _classnames));

        return _react2.default.createElement(
          'div',
          { className: classes },
          _react2.default.createElement(Component, props),
          _react2.default.createElement(
            _tooltip2.default,
            _extends({ className: tooltipClasses }, config),
            text
          )
        );
      };
    };
  };
};