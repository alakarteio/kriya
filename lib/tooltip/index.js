'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _tooltip = require('./tooltip');

Object.defineProperty(exports, 'default', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_tooltip).default;
  }
});

var _hoc = require('./hoc');

Object.defineProperty(exports, 'tooltip', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_hoc).default;
  }
});

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }