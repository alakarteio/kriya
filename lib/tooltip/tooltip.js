'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames2 = require('classnames');

var _classnames3 = _interopRequireDefault(_classnames2);

var _recompose = require('recompose');

var _tooltipStyles = require('../../src/tooltip/tooltip.styles.scss');

var _tooltipStyles2 = _interopRequireDefault(_tooltipStyles);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var Tooltip = function Tooltip(_ref) {
  var style = _ref.style,
      className = _ref.className,
      children = _ref.children,
      show = _ref.show;

  var classes = (0, _classnames3.default)(_tooltipStyles2.default.tooltip, className, _defineProperty({}, _tooltipStyles2.default.show, show));

  return _react2.default.createElement(
    'div',
    { style: style, className: classes },
    children
  );
};

Tooltip.propTypes = {
  style: _propTypes2.default.object,
  className: _propTypes2.default.string,
  children: _propTypes2.default.node,
  show: _propTypes2.default.bool
};

Tooltip.defaultProps = {
  style: {},
  className: '',
  children: undefined,
  show: false
};

exports.default = (0, _recompose.onlyUpdateForPropTypes)(Tooltip);