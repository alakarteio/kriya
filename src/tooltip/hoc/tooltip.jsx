import React from 'react'
import classnames from 'classnames'
import Tooltip from '../tooltip'
import styles from '../../../src/tooltip/hoc/tooltip.styles.scss'

export default (config = {
  left: true,
  top: false,
  right: false,
  bottom: false,
  className: '',
}) => text => Component => (props) => {
  const { left, top, right, bottom, className } = config
  const classes = classnames(
    styles.component,
    className,
  )

  const tooltipClasses = classnames(
    styles.tooltip,
    {
      [styles.left]: left,
      [styles.top]: top,
      [styles.right]: right,
      [styles.bottom]: bottom,
    },
  )

  return (
    <div className={classes}>
      <Component {...props} />
      <Tooltip className={tooltipClasses} {...config} >{text}</Tooltip>
    </div>
  )
}
