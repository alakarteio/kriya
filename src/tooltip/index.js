export { default } from './tooltip'
export { default as tooltip } from './hoc'
