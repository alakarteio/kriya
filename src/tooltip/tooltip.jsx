import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'
import { onlyUpdateForPropTypes } from 'recompose'
import styles from '../../src/tooltip/tooltip.styles.scss'

const Tooltip = ({
  style, className,
  children,
  show,
}) => {
  const classes = classnames(
    styles.tooltip,
    className,
    {
      [styles.show]: show,
    },
  )

  return (
    <div style={style} className={classes}>
      {children}
    </div>
  )
}

Tooltip.propTypes = {
  style: PropTypes.object,
  className: PropTypes.string,
  children: PropTypes.node,
  show: PropTypes.bool,
}

Tooltip.defaultProps = {
  style: {},
  className: '',
  children: undefined,
  show: false,
}

export default onlyUpdateForPropTypes(Tooltip)
