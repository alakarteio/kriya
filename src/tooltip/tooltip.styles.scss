@import '~styles/theme';

.tooltip {
  background: $font-color-dark;
  border-radius: .3em;
  color: $font-color-light;
  opacity: 0;
  padding: .5em 1em;
}

.show {
  opacity: .9;
}
