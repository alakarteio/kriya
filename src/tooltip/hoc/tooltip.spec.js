/* eslint-disable
  import/no-extraneous-dependencies,
  react/jsx-filename-extension,
  react/no-children-prop,
  react/prop-types,
*/
/* eslint-env jest */

import React from 'react'
import snap from 'snap'
import tooltip from './tooltip'

const Child = props => <div title="child">{JSON.stringify(props)}</div>

describe('common/Tooltip/hoc', () => {
  it('should pass config to the tooltip component', () => {
    const Wrapper = tooltip({ left: true })('TEXT')(Child)
    snap(Wrapper)()()
  })

  it('should pass props to the wrapped component', () => {
    const Wrapper = tooltip()('TEXT')(Child)
    snap(Wrapper)({ prop: 'a prop', and: 'an other' })()
  })
})

/* eslint-enable
  import/no-extraneous-dependencies,
  react/jsx-filename-extension,
  react/no-children-prop,
  react/prop-types,
*/
