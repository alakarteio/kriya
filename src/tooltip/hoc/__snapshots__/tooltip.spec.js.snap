// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common/Tooltip/hoc should pass config to the tooltip component 1`] = `
<div
  className="component"
>
  <div
    title="child"
  >
    {}
  </div>
  <div
    className="tooltip tooltip left"
    style={Object {}}
  >
    TEXT
  </div>
</div>
`;

exports[`common/Tooltip/hoc should pass props to the wrapped component 1`] = `
<div
  className="component"
>
  <div
    title="child"
  >
    {"prop":"a prop","and":"an other"}
  </div>
  <div
    className="tooltip"
    style={Object {}}
  >
    TEXT
  </div>
</div>
`;
